src/App.tsx

```diff
@@ -1,10 +1,74 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import {
+  FC,
+  useState,
+  useEffect,
+  useMemo,
+} from 'react';
+import classNames from 'classnames';
+import { Hearts } from 'react-loader-spinner';
 
-const USER_ID = 0;
+import { UserWarning } from './UserWarning';
+import { TodoForm } from './components/TodoForm';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { FilterType } from './types/FilterType';
+import { getActiveTodos, getFilteredTodos } from './helpers/helpers';
+import { ErrorMessage } from './components/ErrorMessage';
+
+const USER_ID = 6928;
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setEerrorMessage] = useState('');
+  const [filterType, setFilterType] = useState(FilterType.All);
+
+  const showError = (errorName: string) => {
+    setEerrorMessage(errorName);
+    setHasError(true);
+
+    setTimeout(() => {
+      setHasError(false);
+    }, 3000);
+  };
+
+  const handleClosingError = () => {
+    setHasError(false);
+  };
+
+  const visibleTodos = useMemo(() => {
+    return getFilteredTodos(todos, filterType);
+  }, [todos, filterType]);
+
+  const activeTodos = useMemo(() => {
+    return getActiveTodos(visibleTodos);
+  }, [visibleTodos]);
+
+  const hasActiveTodos = visibleTodos
+    .some(({ completed }) => !completed);
```

Comment:
I think you can just check if your helper function getActiveTodos return something bigger than 0

Reply:
> I think you can just check if your helper function getActiveTodos returns something bigger than 0

Id need to call function 2 times or write the result in const, think better like this, but thanks)